velox/common/base/tests/StatsReporterTest.cpp

```diff
@@ -119,12 +136,225 @@ TEST_F(StatsReporterTest, trivialReporter) {
+      counterMap.count(kMetricSsdCacheCumulativeWriteCheckpointErrors.str()),
+      1);
+  ASSERT_EQ(counterMap.count(kMetricSsdCacheCumulativeReadSsdErrors.str()), 1);
+  ASSERT_EQ(
+      counterMap.count(kMetricSsdCacheCumulativeReadCheckpointErrors.str()), 1);
+  ASSERT_EQ(counterMap.count(kMetricSsdCacheCachedEntries.str()), 1);
+  ASSERT_EQ(counterMap.count(kMetricSsdCacheCachedRegions.str()), 1);
+  ASSERT_EQ(counterMap.count(kMetricSsdCacheCachedBytes.str()), 1);
+  ASSERT_EQ(counterMap.count(kMetricCacheMaxAgeSecs.str()), 1);
+  ASSERT_EQ(counterMap.count(kMetricMemoryCacheNumAgedOutEntries.str()), 1);
+  ASSERT_EQ(
+      counterMap.count(kMetricMemoryCacheNumCumulativeAgedOutEntries.str()), 1);
+  ASSERT_EQ(counterMap.count(kMetricSsdCacheCumulativeAgedOutEntries.str()), 1);
+  ASSERT_EQ(counterMap.count(kMetricSsdCacheCumulativeAgedOutRegions.str()), 1);
+  // Should not expect any REPORT_IF_NOT_ZERO metrics for the first time.
+  ASSERT_EQ(counterMap.count(kMetricSsdCacheCheckpointsRead.str()), 0);
+  ASSERT_EQ(counterMap.count(kMetricSsdCacheCheckpointsWritten.str()), 0);
+  ASSERT_EQ(counterMap.count(kMetricArbitratorNumRequests.str()), 0);
+  ASSERT_EQ(counterMap.count(kMetricArbitratorNumAborted.str()), 0);
+  ASSERT_EQ(counterMap.count(kMetricArbitratorNumFailures.str()), 0);
+  ASSERT_EQ(counterMap.count(kMetricArbitratorQueueTimeUs.str()), 0);
+  ASSERT_EQ(counterMap.count(kMetricArbitratorArbitrationTimeUs.str()), 0);
+  ASSERT_EQ(counterMap.count(kMetricArbitratorNumShrunkBytes.str()), 0);
+  ASSERT_EQ(counterMap.count(kMetricArbitratorNumReclaimedBytes.str()), 0);
+  ASSERT_EQ(counterMap.count(kMetricArbitratorFreeCapacityBytes.str()), 0);
+  ASSERT_EQ(counterMap.count(kMetricArbitratorNonReclaimableAttempts.str()), 0);
+
+  auto newSsdStats = std::make_shared<cache::SsdCacheStats>();
+  newSsdStats->checkpointsWritten = 10;
+  newSsdStats->checkpointsRead = 10;
+  asyncDataCache.updateStats({.ssdStats = newSsdStats});
+  arbitrator.updateStats(memory::MemoryArbitrator::Stats(
+      10, 10, 10, 10, 10, 10, 10, 10, 10, 10, 10, 10, 10, 10, 10));
+
+  daemon.testingReportOnce();
+  ASSERT_EQ(counterMap.size(), 61);
+  ASSERT_EQ(counterMap.count(kMetricSsdCacheCheckpointsRead.str()), 1);
+  ASSERT_EQ(counterMap.count(kMetricSsdCacheCheckpointsWritten.str()), 1);
+  ASSERT_EQ(counterMap.count(kMetricArbitratorNumRequests.str()), 1);
+  ASSERT_EQ(counterMap.count(kMetricArbitratorNumAborted.str()), 1);
+  ASSERT_EQ(counterMap.count(kMetricArbitratorNumFailures.str()), 1);
+  ASSERT_EQ(counterMap.count(kMetricArbitratorQueueTimeUs.str()), 1);
+  ASSERT_EQ(counterMap.count(kMetricArbitratorArbitrationTimeUs.str()), 1);
+  ASSERT_EQ(counterMap.count(kMetricArbitratorNumShrunkBytes.str()), 1);
+  ASSERT_EQ(counterMap.count(kMetricArbitratorNumReclaimedBytes.str()), 1);
+  ASSERT_EQ(counterMap.count(kMetricArbitratorFreeCapacityBytes.str()), 1);
+  ASSERT_EQ(counterMap.count(kMetricArbitratorNonReclaimableAttempts.str()), 1);
```

Comment:
We could also wait for sufficient long time to make sure that the stats has been reported by the background daemon? Thanks!